### PR TITLE
Rename TracerPlugin to TelemetryPlugin 

### PR DESCRIPTION
**Pull request by pgomulka**
with the support of metrics the TracerPlugin name is no longer adequate. Renaming this to TelemetryPlugin.
Also introducing TelemetryProvider interface. While it is only used in Node.java at the moment to fetch Tracer instance, it is intended to be used in Plugin::createComponents (to be done in separate commit due to
the broad scope of this method)
This will allow for plugins to get access to both Tracer and Metric interfaces
without the need to add yet another argument to createComponents

Also adding internal subpackage in module/apm so that it is more obvious
which packages are not exported

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/node/Node.java [hunk 1]
```diff
@@ -1287,14 +1289,14 @@ private static ReloadablePlugin wrapPlugins(List<ReloadablePlugin> reloadablePlu
...
 
-    private Tracer getTracer(PluginsService pluginsService, Settings settings) {
-        final List<TracerPlugin> tracerPlugins = pluginsService.filterPlugins(TracerPlugin.class);
+    private TelemetryProvider getTelemetryProvider(PluginsService pluginsService, Settings settings) {
+        final List<TelemetryPlugin> telemetryPlugins = pluginsService.filterPlugins(TelemetryPlugin.class);
 
-        if (tracerPlugins.size() > 1) {
-            throw new IllegalStateException("A single TracerPlugin was expected but got: " + tracerPlugins);
+        if (telemetryPlugins.size() > 1) {
+            throw new IllegalStateException("A single TracerPlugin was expected but got: " + telemetryPlugins);
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/node/Node.java:1296]: `"A single TracerPlugin was expected but got: "` 
-> 
`"A single TelemetryPlugin was expected but got: "`

**Comment by stu-elastic** [comment]
Some comments in `APMJvmOptions` also need to reflect `telemetry` instead of `tracing`.

**Review by stu-elastic — APPROVED** [review]
lgtm with comments addressed

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
bwc test failure unrelated
https://github.com/elastic/elasticsearch/issues/99778

**Pull request merged**